perllib/FixMyStreet/Roles/ContactExtra.pm

```diff
@@ -25,8 +25,10 @@ sub for_bodies {
 }
 
 sub by_categories {
-    my ($rs, $area_id, @contacts) = @_;
-    my %body_ids = map { $_->body_id => 1 } FixMyStreet::DB->resultset('BodyArea')->search({ area_id => $area_id });
+    my ($rs, $area_id, $contacts, $bodies) = @_;
```

Comment:
So $area_id isn't used if $bodies is provided? I think this would be clearer if you passed in a hash of arguments (so area_id *or* bodies), then bodies is set from area_id if not given.